test/unit/test_diskfile.py

```diff
@@ -15,234 +15,416 @@
 
 """Tests for swift_scality_backend.diskfile"""
 
+import itertools
 import httplib
-import logging
+import StringIO
 import unittest
 import urllib
 
+
 import eventlet
 import mock
 import swift.common.exceptions
+import swift.common.utils
 
-import swift_scality_backend.diskfile
-import swift_scality_backend.utils
-from swift_scality_backend.exceptions import SproxydConfException
-
-_metadata = {}
-CONN_TIMEOUT = 1
-PROXY_TIMEOUT = 1
-
+NEW_SPLICE = 'new_splice'
+OLD_SPLICE = 'old_splice'
+NO_SPLICE_AT_ALL = 'no_splice_at_all'
+try:
+    import swift.common.splice
+    SPLICE = NEW_SPLICE
+except ImportError:
+    if hasattr(swift.common.utils, 'system_has_splice'):
+        SPLICE = OLD_SPLICE
+    else:
+        SPLICE = NO_SPLICE_AT_ALL
 
-def _mock_get_meta(self, obj):
-    """get the metadata from a fake dictionary."""
-    return _metadata
 
+from swift_scality_backend.diskfile import SproxydFileSystem, DiskFileWriter, \
+    DiskFileReader, DiskFile
+from swift_scality_backend.exceptions import SproxydConfException, \
+    SproxydHTTPException
 
-def fake_put_meta(name, metadata):
-    """set the metadata to a fake dictionary."""
-    _metadata[name] = metadata
 
+class FakeHTTPResp(httplib.HTTPResponse):
 
-class FakeHTTPResponse:
-    """fake response class for faking HTTP answers."""
-    def __init__(self):
-        self.status = 0
-        self.msg = ""
-        self.reason = ""
+    def __init__(self, status=200):
+        self.status = status
+        self.reason = 'because'
+        # This is base64.b64encode(pickle.dumps("fake"))
+        headers = StringIO.StringIO('x-scal-usermd: UydmYWtlJwpwMAou')
```

Comment:
Rather than putting a static value and some comment explaining what the value is, just calculate it?
```
headers = StringIO.StringIO('x-scal-usermd: %s' % base64.b64encode((pickle.dumps('fake')))
```